### ISSUE TITLE
:adapter from config files - Postgres

### ISSUE THREAD

**Issue by hawksight** [phoenixframework/phoenix_ecto#106]
Building some applications today I've noticed that both 3.5.0 and 4.0.0 of `:phoenix_ecto` return an error when trying to test the application:

```
warning: retrieving the :adapter from config files for Transfer.Repo is deprecated.
Instead pass the adapter configuration when defining the module:

    defmodule Transfer.Repo do
      use Ecto.Repo,
        otp_app: :transfer,
        adapter: Ecto.Adapters.Postgres

  lib/ecto/repo/supervisor.ex:100: Ecto.Repo.Supervisor.deprecated_adapter/3
  lib/ecto/repo/supervisor.ex:64: Ecto.Repo.Supervisor.compile_config/2
  lib/transfer/repo.ex:2: (module)
  (elixir) src/elixir_compiler.erl:85: :elixir_compiler.dispatch/6
  (elixir) src/elixir_module.erl:223: :elixir_module.eval_form/6


== Compilation error in file lib/transfer/repo.ex ==
** (ArgumentError) adapter Ecto.Adapters.Postgres was not compiled, ensure it is correct and it is included as a project dependency
    lib/ecto/repo/supervisor.ex:71: Ecto.Repo.Supervisor.compile_config/2
    lib/transfer/repo.ex:2: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
```

Repo file:
```
defmodule Transfer.Repo do
  use Ecto.Repo, otp_app: :transfer
end
```

Config changes per environment, but essentially for dev
```
# Configure your database
config :transfer, Transfer.Repo,
  adapter: Ecto.Adapters.Postgres,
  url: System.get_env("DATABASE_URL") || "postgres://localhost/transfer",
  pool_size: String.to_integer(System.get_env("POOL_SIZE") || "2"),
  loggers: [Ecto.LogEntry]
```

I tried the config as described in the deprecation warning but got the following issues:
```
== Compilation error in file lib/transfer/repo.ex ==
** (ArgumentError) adapter Ecto.Adapters.Postgres was not compiled, ensure it is correct and it is included as a project dependency
    lib/ecto/repo/supervisor.ex:71: Ecto.Repo.Supervisor.compile_config/2
    lib/transfer/repo.ex:2: (module)
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
make: *** [test] Error 1
```

Also notice in `mix deps.get` for 3.5.0:
```
  phoenix_ecto 3.5.0 RETIRED!
    (invalid)
```
Hence tried version 4.0.0 with no luck.

Solution: `{:phoenix_ecto, "~> 3.4.0"},` which works as usual.

Might this be similar to #105  where :adapter config is not accepted? Except in this case its for Postgres.

**Comment by josevalim** [comment]
You need to add `ecto_sql` as a dependency if you are trying phoenix_ecto 4.0.0.

**Comment by josevalim** [comment]
But to clarify, there is a bug on our side here too.

**Comment by josevalim** [comment]
Fixed by releasing v3.6.0. A fresh Phoenix v1.3.x install should no longer have this error. thanks!

**Comment by hawksight** [comment]
Thanks for the upgrade tip on 4.0.0.
Just checked 3.6.0 and it works perfectly, thank you.
